### PR TITLE
Memory format propagation parser patch

### PR DESCRIPTION
**Pull request by jjsjann123**
We forgot to mark memory format on a few ops (clamp/softplus/threshold)

This causes us to output tensor in wrong shape, causing failure on torchvision models https://github.com/pytorch/pytorch/issues/76791

### CODE REVIEW

**Comment by jjsjann123** [comment]
cc'ing @davidberard98

**Review by rdspring1 — APPROVED** [review]
LGTM

File: test/test_jit_cuda_fuser.py [hunk 1]
```diff
@@ -2180,6 +2180,33 @@ def t(x, y):
...
+        sizes = [2, 3, 4, 5]
+        dtype = torch.float
+        device = "cuda"
+        x = torch.randn(sizes, dtype=dtype, device=device).to(memory_format=torch.channels_last)
+        y = torch.randn(sizes, dtype=dtype, device=device).to(memory_format=torch.channels_last)
+        z = torch.randn(sizes, dtype=dtype, device=device).to(memory_format=torch.channels_last)
+
+        def t(x, y, w):
+            tmp = torch.lerp(x, y, w)
+            tmp = torch.clamp(x, -1.0, 0.5)
```
Comment (rdspring1) [inline test/test_jit_cuda_fuser.py]: Will the `lerp` output be overridden by `clamp` here?

File: test/test_jit_cuda_fuser.py [hunk 2]
```diff
@@ -2180,6 +2180,33 @@ def t(x, y):
...
+            tmp = torch.nn.functional.softplus(tmp)
+            return torch.threshold(tmp, -2.0, 0.5)
+
+        t_jit = torch.jit.script(t)
+        with nvfuser_singleton_fusion(True):
```
Comment (rdspring1) [inline test/test_jit_cuda_fuser.py]: Do we need to enable `nvfuser_singleton_fusion` if function `t` has multiple operations?

File: test/test_jit_cuda_fuser.py [hunk 3]
```diff
@@ -2180,6 +2180,33 @@ def t(x, y):
...
+            for _ in range(5):
+                jit_o = t_jit(x, y, z)
+
+        o = t(x, y, z)
```
Comment (rdspring1) [inline test/test_jit_cuda_fuser.py]: Could we use `_run_helper` here if we add a stride check?

File: test/test_jit_cuda_fuser.py (hunk 1 above)
Comment (jjsjann123) [reply test/test_jit_cuda_fuser.py]: good catch! that's not intended...

File: test/test_jit_cuda_fuser.py (hunk 2 above)
Comment (jjsjann123) [reply test/test_jit_cuda_fuser.py]: Nope we don't.... Should be more careful with my copy paste

File: test/test_jit_cuda_fuser.py (hunk 3 above)
Comment (jjsjann123) [reply test/test_jit_cuda_fuser.py]: That's very true. I can add a new `_run_helper_validate_stride` to handle that

**Comment by jjsjann123** [comment]
passing tests~ I'm merging this one after CI.

**Pull request merged**